modules/math/src/main/java/org/pixel/math/Rectangle.java

```diff
@@ -459,18 +460,21 @@ public void set(Rectangle other) {
     }
 
     @Override
-    public String toString() {
-        return String.format("%s: [x: %f, y: %f, w: %f, h: %f]",
-                this.getClass().getSimpleName(), getX(), getY(), getWidth(), getHeight());
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Rectangle)) return false;
+        Rectangle rectangle = (Rectangle) o;
+        return Float.compare(rectangle.x, x) == 0 && Float.compare(rectangle.y, y) == 0 && Float.compare(rectangle.width, width) == 0 && Float.compare(rectangle.height, height) == 0;
```

Comment:
Keep the previous equals body (add the this == 0). Float.compare is heaver than checking the values (which in a game-oriented library is important)

Reply:
My bad, I created this equals function because it didn't exist at the time